TypeSystem/types.h

```diff
@@ -41,6 +41,13 @@ struct CurrentSuper {
   virtual ~CurrentSuper() = default;
 };
 
+// For `unique_ptr<>`-s.
+struct CurrentSuperDeleter {
+  void operator()(CurrentSuper* ptr) {
+    delete ptr;
+  }
+};
```

Comment:
The only place `CurrentSuperDeleter` is mentioned in this PR is this file. Is it for future use or some legacy?